### PR TITLE
fix: getCodeFixesAtPosition fails for TS2339 for specific input with large pojo

### PR DESCRIPTION
**Pull request by wondersloth**
This fixes #1071

This PR fixes an issue when rehearsal attempts to `getCodeFixAtPosition` on a file where the base class can not be determined due to import semantics.

Example Input:

```
import { Base } from 'some-package';

export class Something extends Base {
  constructor() {
    super();
    this.setRoutes({
      'enterprise-cookies': '',
      'feed.update': '',
      'help.ads': '',
      'help.binding-benefits': '',
      'help.certificates': '',
      'help.disability-answer-desk': '',
      'help.instructor-analytics': '',
      'help.learning': '',
      'help.learning-privacy': '',
      'help.learning-progress': '',
      'help.learning-teams': '',
      'help.china': '',
      'help.likes': '',
      'help.lms-completion': '',
      'help.lynda-subscriber': '',
      'help.questions-privacy': '',
      'help.reviews': '',
      'help.share-certificate': '',
      'help.sync-learning-history': '',
    });
  }
}
```
Note: If there is only a couple properties, TypeScript does not error. Though in our use case somewhere around 13 items or more, it starts to freak out.

### Details

- Adds DEBUG to `codefixes` so it will be easier to quickly debug these types of errors (First commit).
- Fixes bug #1070 has reproducible use cases. 
- Creates test for existing error suppression logic.
- Allows for future extension of additional suppressions more easily.

### Why does this have to be fixed?
If we don't suppress this error, any class that extends from a base class from a non-typescript converted package will fail in rehearsal.

### Should we fix this upstream in TypeScript
Yes, but now.

I will be following this up later with a new TS issue.

At this time, I don’t think we should invest time to fix this upstream in TypeScript, as this fix will unblock our internal migrations.

### CODE REVIEW

File: packages/codefixes/src/typescript-codefix-collection.ts [hunk 1]
```diff
@@ -14,12 +15,27 @@ import { Diagnostics } from './diagnosticInformationMap.generated.js';
...
 
+const DEBUG_CALLBACK = debug('rehearsal:codefixes:TypeScriptCodeFixCollection');
+
 const __filename = fileURLToPath(import.meta.url);
 const __dirname = dirname(__filename);
 const require = Module.createRequire(import.meta.url);
 
 const { SemicolonPreference, getDefaultFormatCodeSettings } = ts;
 
+type SupressedError = { code: number; message: string };
```
Comment (lynchbomb) [inline packages/codefixes/src/typescript-codefix-collection.ts:26]: `SuppressedError`

File: packages/codefixes/src/typescript-codefix-collection.ts [hunk 2]
```diff
@@ -14,12 +15,27 @@ import { Diagnostics } from './diagnosticInformationMap.generated.js';
...
+
+const SUPRESSED_ERRORS: Array<SupressedError> = [
```
Comment (lynchbomb) [inline packages/codefixes/src/typescript-codefix-collection.ts:28]: `SUPPRESSED_ERRORS:`

File: packages/codefixes/src/typescript-codefix-collection.ts [hunk 3]
```diff
@@ -57,13 +73,23 @@ export class TypescriptCodeFixCollection implements CodeFixCollection {
         userPreferences
       );
     } catch (e) {
-      const hideError =
-        diagnostic.code == Diagnostics.TS2345.code &&
-        e instanceof TypeError &&
-        e.message.includes(`Cannot read properties of undefined (reading 'flags')`);
+      const code = diagnostic.code;
+      const supressedFound = SUPRESSED_ERRORS.find((d) => d.code == code);
```
Comment (lynchbomb) [inline packages/codefixes/src/typescript-codefix-collection.ts:77]: suppressedFound

File: packages/codefixes/src/typescript-codefix-collection.ts [hunk 4]
```diff
@@ -57,13 +73,23 @@ export class TypescriptCodeFixCollection implements CodeFixCollection {
...
+
+      const supressError: boolean =
```
Comment (lynchbomb) [inline packages/codefixes/src/typescript-codefix-collection.ts:79]: suppressError

File: packages/codefixes/test/__snapshots__/supressed-errors.test.ts.snap [hunk 5]
```diff
@@ -0,0 +1,48 @@
+// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html
+
+exports[`Supressed Errors > TS2339: False expression: Token end is child end, supressed-errors/missingMethod.ts 1`] = `
+"// @ts-expect-error @rehearsal TODO TS2307: Cannot find module 'some-package' or its corresponding type declarations.
+import { Base } from \\"some-package\\";
```
Comment (lynchbomb) [inline packages/codefixes/test/__snapshots__/supressed-errors.test.ts.snap:5]: wondering in the instance when ts *can* find the module `some-package` are we still seeing this error?

**Review by lynchbomb — CHANGES_REQUESTED** [review]
couple minor things. once tests are passing this should be good

File: packages/codefixes/test/__snapshots__/supressed-errors.test.ts.snap (hunk 5 above)
Comment (wondersloth) [reply packages/codefixes/test/__snapshots__/supressed-errors.test.ts.snap]: In learning-web, I updated a tsconfig to resolve imports to that package, and it still failed. Despite being able to click through the package name to get to the implementation.

This problem is less about the actual class and something breaking given the method not found on the class, and the large size of the pojo.

**Review by lynchbomb — APPROVED** [review]

**Pull request merged**